oddt/fingerprints.py

```diff
@@ -541,6 +549,71 @@ def combinatorial_rmsd(reference, query):
         return np.sqrt((numla / nula) * (numpa / nupa))
 
 
+def PLECFP(ligand, protein, depth_ligand=2, depth_protein=4, distance_cutoff=4.5,
+           size=16384, count_bits=True, sparse=True):
+    """Protein ligand extended connectivity fingerprint. For every pair of
+    atoms in contact, compute ECFP and then hash every single, corresponding depth.
+
+    Parameters
+    ----------
+    ligand, protein : oddt.toolkit.Molecule object
+            Molecules, which are analysed in order to find interactions.
+    depth_ligand, depth_protein : int (deafult = (2, 4))
+        The depth of the fingerprint, i.e. the number of bonds in Morgan
+        algorithm. Note: For ECFP2: depth = 1, ECFP4: depth = 2, etc.
+    size: int (default = 16384)
+        SPLIF is folded to given size.
+    distance_cutoff: float (default=4.5)
+        Cutoff distance for close contacts.
+    sparse : bool (default = True)
+        Should fingerprints be dense (contain all bits) or sparse (just the on
+        bits).
+    count_bits : bool (default = True)
+        Should the bits be counted or unique. In dense representation it
+        translates to integer array (count_bits=True) or boolean array if False.
+
+    Returns
+    -------
+    PLECFP : numpy array
+        Calculated fp (size = no. of atoms in contacts * max(depth_protein, depth_ligand))
+
+    """
+    result = []
+    # removing h
+    protein_dict = protein.atom_dict[protein.atom_dict['atomicnum'] != 1]
+    ligand_dict = ligand.atom_dict[ligand.atom_dict['atomicnum'] != 1]
+
+    # atoms in contact
+    protein_atoms, ligand_atoms = close_contacts(
+        protein_dict, ligand_dict, cutoff=distance_cutoff)
+
+    for i, (ligand_atom, protein_atom) in enumerate(zip(ligand_atoms, protein_atoms)):
+        ligand_ecfp = _ECFP_atom_hash(ligand, int(ligand_atom['id']), depth=depth_ligand)
+        protein_ecfp = _ECFP_atom_hash(protein, int(protein_atom['id']), depth=depth_protein)
+        assert len(ligand_ecfp) == depth_ligand + 1
+        assert len(protein_ecfp) == depth_protein + 1
+        # fill_value is parameter from zip_longest, it's used when two arrays are not same shape
+        # e.g fp1='xyz', fp2='abcde' returns 'xa', 'yb', 'zc', 'zd', 'ze'
+        # so if one list is shorter, we use the last element
+        if depth_ligand < depth_protein:
+            fillvalue = ligand_ecfp[-1]
+        else:
+            fillvalue = protein_ecfp[-1]
+        for pair in zip_longest(ligand_ecfp, protein_ecfp, fillvalue=fillvalue):
+                result.append(hash32(pair))
+    # folding and sorting
+    plecfp = np.sort(fold(np.array(result), size=size))
+
+    # count_bits
+    if not count_bits:
+        plecfp = np.unique(plecfp)
+
+    # sparse or dense FP
+    if not sparse:
+        plecfp = sparse_to_dense(plecfp.astype(int), size=size)
```

Comment:
`astype(int)` is redundant, since fold does casting to int. Check out the new version.